test/xapi/problem_test.rb

```diff
@@ -18,6 +18,21 @@ def test_existing_problem
     assert_equal 'http://example.com', problem.source_url
   end
 
+  def test_existing_problem_in_new_location
```

Comment:
This test is testing too much.

Try and work out what it is you are trying to test and only test that.